### DISCUSSION TITLE
Make `wait-for` wait for full startup of vrnetlab containers

### DISCUSSION THREAD

**Discussion by JakeBednard** [srl-labs/containerlab#1048] [Ideas]
FR: Make `wait-for` wait for full startup of vrnetlab containers

One FR that I have is that `wait-for` cannot sense the full startup of vrnetlab containers. The Dependency Manager senses these containers as started at the `level=debug msg="Container started: \"my-container\""` log message. This does not reflect that the vrnetlab containers utilize a full bootup and load sequence on top of qemu, before becoming ready. This can be problematic in resource constrained environments. 

One solution would be to utilize [launch.py](https://github.com/hellt/vrnetlab/blob/master/n9kv/docker/launch.py) to set a flag within the container after the bootup and configuration apply has been completed. Then the Dependency Manager would need to watch for the flag-flip before marking the dependency as started. I assume this feature would need to signify it presence in the deploy.clab.yml by creating a new node-type or flag to signal the need to watch the flag-flip.

Thanks for adding the `wait-for` feature @steiler!

**Comment by steiler** [comment]
Hi @JakeBednard,
I've realized already, that there might be different kinds of dependencies between nodes. See: #1027 

Also tried to quickly implement phases like configured (post-deploy) and even healthy (considering runtime health checks).
However, containerlab today runs in parallel for pre-deployment, create and start phase, but then waits for all threads to complete before starting the configuration phase.
Hence introducing dependencies like _when x is configured / health start y_ will result in a deadlock.
If there is enough request, sure I can digg a little deeper and make this all happen. For now, without real demand that seemed to be too much of an effort.
https://github.com/srl-labs/containerlab/issues/1027#issuecomment-1259213291
